### PR TITLE
undervolt: expose power limits as Nixopts

### PR DESCRIPTION
**Pull request by Atemu**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by pbogdan — APPROVED** [review]

File: nixos/modules/services/hardware/undervolt.nix
Comment (glittershark) [inline nixos/modules/services/hardware/undervolt.nix:31]: why expose these as `p1` and `p2` rather than just reusing the argument name from the underlying tool?

Comment (Atemu) [reply nixos/modules/services/hardware/undervolt.nix:31]: The names of the short option are p1 & p2 which are also the terms used in hardware forums etc. I don't think I've ever heard 'long' & 'short' for these.

File: nixos/modules/services/hardware/undervolt.nix
Comment (glittershark) [inline nixos/modules/services/hardware/undervolt.nix:140]: just a thought, but we might be able to use a submodule type to deduplicate the `window` and `limit` type definitions. But that might be overly abstracted, so I'll leave it up to you / other reviewers

**Review by glittershark — APPROVED** [review]

File: nixos/modules/services/hardware/undervolt.nix
Comment (timokau) [reply nixos/modules/services/hardware/undervolt.nix:31]: I agree that we should use the original name of the option. Its not our place to rename it, it only adds to the confusion. If you mention the term "P1" and "P2" in the relevant descriptions, searchability should be just as good.

If feel strongly that the names should be kept p1 and p2, the description should at least mention the name of the underlying configuration options.

**Review by timokau — CHANGES_REQUESTED** [review]

File: nixos/modules/services/hardware/undervolt.nix
Comment (Atemu) [reply nixos/modules/services/hardware/undervolt.nix:31]: These are the original names of the options: https://github.com/georgewhewell/undervolt/blob/13fa33d05ae1d1306a2d3e2cd921028c1eb785d6/undervolt.py#L376

**Review by timokau — APPROVED** [review]
Thanks @glittershark and @pbogdan for the reviews.

**Comment by Atemu** [comment]
Thank you!

Should we backport this to 20.09? It already has the undervolt update which allows for this feature.

**Comment by timokau** [comment]
Yes, I think that would be fine, since its purely a feature-addition without risk of breakge. Its not my decision though. I think it would be best to open a backport PR and ping one of the release managers.

**Pull request merged**
